homeassistant/components/plant.py

```diff
@@ -192,28 +209,67 @@ def _update_state(self):
             params = self.READINGS[sensor_name]
             value = getattr(self, '_{}'.format(sensor_name))
             if value is not None:
-                if 'min' in params and params['min'] in self._config:
-                    min_value = self._config[params['min']]
-                    if value < min_value:
-                        result.append('{} low'.format(sensor_name))
-                        self._icon = params['icon']
-
-                if 'max' in params and params['max'] in self._config:
-                    max_value = self._config[params['max']]
-                    if value > max_value:
-                        result.append('{} high'.format(sensor_name))
-                        self._icon = params['icon']
+                if sensor_name == READING_BRIGHTNESS:
+                    result.append(self._check_min(
+                        sensor_name, self._brightness_history.max, params))
+                else:
+                    result.append(self._check_min(sensor_name, value, params))
+                result.append(self._check_max(sensor_name, value, params))
+
+        result = filter(lambda x: x is not None, result)
```

Comment:
Don't use `filter`.

```python
result = [entry for entry in result if entry is not None]
```

Reply:
ACK.

In case anyone is interested why: https://stackoverflow.com/questions/3013449/list-filtering-list-comprehension-vs-lambda-filter